### PR TITLE
Avoid adding ipfs dependencies to `github.com/containerd/stargz-snapshotter`

### PR DESCRIPTION
**Pull request by ktock**
The module `github.com/containerd/stargz-snapshotter` is used by various projects (e.g. BuildKit, k3s, CernVM-FS, ...). We should keep the dependency as minimal as possible.

Though IPFS-related logic is experimental and optional as of now, it contains [lots of dependencies](https://github.com/containerd/stargz-snapshotter/commit/99edafb9cb051d0c094fa91dd1da3caccc77b0fc#diff-3295df7234525439d778f1b282d146a4f1ff6b415248aaac074e8042d9f42d63). We should avoid adding them to `github.com/containerd/stargz-snapshotter`.

This commit solves this issue by introducing new sub modules for ipfs-related logic.

- `ipfs`: IPFS-related plugins for containerd and stargz-snapshotter
- `cmd`: Dependencies of binary commands.

Additionally, ipfs-related logic is eliminated from `github.com/containerd/stargz-snapshotter/fs` by making the remote resolver logic configurable. IPFS dependencies are included in `ipfs` and `cmd` but not in `github.com/containerd/stargz-snapshotter`.

This also eliminates the dependency to `github.com/containerd/containerd/api` from `github.com/containerd/stargz-snapshotter`. This will solve https://github.com/containerd/stargz-snapshotter/pull/508.

### CODE REVIEW

File: ipfs/go.sum [hunk 1]
```diff
@@ -0,0 +1,1771 @@
...
+github.com/gxed/hashland/murmur3 v0.0.1/go.mod h1:KjXop02n4/ckmZSnY2+HKcLud/tcmvhST0bie/0lS48=
+github.com/hashicorp/errwrap v0.0.0-20141028054710-7554cd9344ce/go.mod h1:YH+1FKiLXxHSkmPseP+kNlulaMuP3n2brvKWEqk/Jc4=
+github.com/hashicorp/errwrap v1.0.0/go.mod h1:YH+1FKiLXxHSkmPseP+kNlulaMuP3n2brvKWEqk/Jc4=
+github.com/hashicorp/go-multierror v0.0.0-20161216184304-ed905158d874/go.mod h1:JMRHfdO9jKNzS/+BTlxCjKNQHg/jZAft8U7LloJvN7I=
+github.com/hashicorp/go-multierror v1.0.0/go.mod h1:dHtQlpGsu+cZNNAkkCN/P3hoUDHhCYQXV3UM06sGGrk=
+github.com/hashicorp/golang-lru v0.5.0/go.mod h1:/m3WP610KZHVQ1SGc6re/UDhFvYD7pJ4Ao+sR/qLZy8=
+github.com/hashicorp/golang-lru v0.5.1/go.mod h1:/m3WP610KZHVQ1SGc6re/UDhFvYD7pJ4Ao+sR/qLZy8=
+github.com/hashicorp/golang-lru v0.5.3/go.mod h1:iADmTwqILo4mZ8BN3D2Q6+9jd8WM5uGBxy+E8yxSoD4=
+github.com/hashicorp/golang-lru v0.5.4 h1:YDjusn29QI/Das2iO9M0BHnIbxPeyuCHsjMW+lJfyTc=
+github.com/hashicorp/golang-lru v0.5.4/go.mod h1:iADmTwqILo4mZ8BN3D2Q6+9jd8WM5uGBxy+E8yxSoD4=
```
Comment (AkihiroSuda) [inline ipfs/go.sum:448]: Replace this with your fork

Comment (ktock) [reply ipfs/go.sum]: Fixed.

**Review by AkihiroSuda — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
Can we have a new release soon?

**Comment by AkihiroSuda** [comment]
(Before that we should update the containerd lib and dedupe the converter pkg)

**Pull request merged**